### PR TITLE
Taylans changes on top of rebase

### PR DESCRIPTION
**Pull request by taylanbil**

### CODE REVIEW

File: fairseq/criterions/wav2vec_criterion.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def __init__(self, task, infonce=False, loss_weights=None, log_keys=None):
         self.loss_weights = loss_weights
         self.log_keys = [] if log_keys is None else log_keys
 
-    def forward(self, model, sample, reduce=True, log_pred=False):
```
Comment (taylanbil) [inline fairseq/criterions/wav2vec_criterion.py:44]: not used.

File: fairseq/data/bucket_pad_length_dataset.py [hunk 2]
```diff
@@ -37,24 +37,7 @@ def __init__(
...
-        def get_bucketed_sizes(orig_sizes, buckets):
-            sizes = np.copy(orig_sizes)
-            assert np.min(sizes) >= 0
-            start_val = -1
-            for end_val in buckets:
-                mask = (sizes > start_val) & (sizes <= end_val)
-                sizes[mask] = end_val
-                start_val = end_val
-            return sizes
-
```
Comment (taylanbil) [inline fairseq/data/bucket_pad_length_dataset.py:57]: imported from data_utils.

File: fairseq/models/wav2vec/wav2vec2.py [hunk 3]
```diff
@@ -534,7 +534,9 @@ def forward(
             y = self.project_q(y)
 
             if self.negatives_from_everywhere:
-                neg_cands = (self.quantizer(unmasked_features, produce_targets=False))["x"]
+                neg_cands = self.quantizer(
+                    unmasked_features, produce_targets=False
+                )["x"]
```
Comment (taylanbil) [inline fairseq/models/wav2vec/wav2vec2.py:539]: style

File: fairseq_cli/train.py [hunk 4]
```diff
@@ -176,19 +176,6 @@ def is_better(a, b):
...
-
-    xm.rendezvous("tpu_data_loader")  # wait for all workers
-    device = utils.get_tpu_device(args)
-    return iterators.CountingIterator(
-        pl.ParallelLoader(itr, [device]).per_device_loader(device),
-        start=getattr(itr, "n", 0),
-        total=len(itr),
-    )
-
-
```
Comment (taylanbil) [inline fairseq_cli/train.py:191]: this is slightly different from the one in `utils.py`, any use for this one?

**Comment by bilgeacun** [comment]
looks good

**Review by bilgeacun — APPROVED** [review]

**Pull request merged**